typed-racket-more/typed/web-server/http.rkt

```diff
@@ -57,65 +58,103 @@
                        [#:struct response
                                  ([code : Natural]
                                   [message : Bytes]
-                                  [seconds : Natural]
+                                  [seconds : Real]
                                   [mime : (Option Bytes)]
                                   [headers : (Listof Header)]
-                                  [output : (-> Output-Port Void)])]
+                                  [output : (-> Output-Port Any)])]
                        [response/full
                         (-> Natural
                             Bytes
-                            Number
+                            Real
                             (Option Bytes)
                             (Listof Header)
                             (Listof Bytes)
                             Response)]
                        [response/output
-                        (-> (-> Output-Port Void)
+                        (-> (-> Output-Port Any)
                             [#:code Natural]
                             [#:message Bytes]
-                            [#:seconds Number]
+                            [#:seconds Real]
                             [#:mime-type (Option Bytes)]
                             [#:headers (Listof Header)]
                             Response)]
                        [TEXT/HTML-MIME-TYPE Bytes])
 
+;; For backwards compatability:
+;; these bindings are not exported by web-server/http,
+;; but they were exported by previous versions of
+;; web-server/http based on net/cookie.
```

Comment:
Not relevant to this PR, but was this a backwards-incompatible change to `web-server/http` in the past?

Reply:
If you mean the exported names: no. Line 86 should have said `;; typed/web-server/http based on net/cookie.` (I will submit a correction shortly.) The names `cookie?`, `cookie-name?`, `cookie-value?`, and `valid-domain?` are not and never were provided by untyped `web-server/http`: they were only provided by `typed/web-server/http`.

If you mean about the underlying change: this was discussed at <https://github.com/racket/web-server/issues/20>. IIRC the conclusion was basically that `web-server/http` had tried very hard to avoid making promises about its internal representation of cookies (e.g. it did not provide `cookie?`). So we took pains to keep the `web-server/http` API compatible (e.g. parsing dates provided as strings …), but ultimately wanted to track the underlying change in the web.

Reply:
The fact that `typed/web-server/http` previously provided a _different_ opaque `Cookie` type than `typed/net/cookie` (the deprecated one) seems to be consistent with that rationale.

Reply:
@samth One difference I just thought of is that previously, the predicates could be used for occurrence typing (e.g. `cookie-name?` for `Cookie-Name`). The actual representation hasn't changed (they were and are still `String`s, with slightly different well-formedness checks), but that might make some programs now fail to typecheck. I can add little wrappers to attach a `String` proposition.